epilepsy12/management/commands/write_azure_pg_service_file.py

```diff
@@ -0,0 +1,50 @@
+import os
+import requests
+from django.core.management.base import BaseCommand
+
+# https://www.postgresql.org/docs/current/libpq-pgservice.html
+#
+# Allows us to periodically update the access token for Entra ID
+# authenticated connections in Azure (triggering this command from cron).
+# 
+# Django specifically does not allow you to update settings at
+# runtime so we store the connection details in a separate file.
+
+class Command(BaseCommand):
+    help = "Generate a Postgres service file to use an Azure managed identity to authenticate with the database."
+    
+    def fetch_azure_token(self):
+        url = 'http://169.254.169.254/metadata/identity/oauth2/token'
+
+        params = {
+            'api-version': '2018-02-01',
+            'resource': 'https://ossrdbms-aad.database.windows.net'
```

Comment:
It turns out the `client_id` parameter was optional as we only have a single managed identity attached to the VMs